calendar/month.go

```diff
@@ -289,12 +339,48 @@ func (m *monthView) getCalendarDay(day time.Time, x, y int) []string {
 		}
 	}
 
+	// TODO: figure out how to render event over multiple days
+
+	var holidayRow string = "       "
+
+	var eventRows = []string{"", "", ""}
+
+	for index, event := range events {
+		var eventStr string
+
+		// we only have 3 rows
+		if index > 3 {
+			break
+		}
+
+		// starts and stops at the same day
+		if dateEqual(day, event.startTime) && dateEqual(day, event.stopTime) {
+			eventStr = event.text[:11] + ".."
+		} else if dateEqual(day, event.startTime) {
+			// print text?
+			eventStr = event.text[:13]
+		} else if dateEqual(day, event.stopTime) {
+			strLength := len(event.text)
+			if strLength > 13 {
+				eventStr = event.text[13:] // we need to figure out how long the text is
+			}
+		}
+
+		eventRows[index] = fmt.Sprintf("[%s]%s[-:-:-]", event.color, ensureLength(eventStr, 11))
+	}
+
 	dayStr := fmt.Sprintf(
-		"│[%s] [%s]%2d[-:-][%s]        [-:-]\n"+
-			"│[%s]           [-:-]\n"+
-			"│[%s]           [-:-]\n"+
-			"│[%s]           [-:-]",
-		cursor, dayNumberColor, day.Day(), cursor, cursor, cursor, cursor,
+		"│[%[1]s] [%[2]s]%2[3]d[-:-][%[1]s] %[4]s[-:-]\n"+
+			"│[%[1]s]%[5]s[-:-]\n"+
+			"│[%[1]s]%[6]s[-:-]\n"+
+			"│[%[1]s]%[7]s[-:-]",
+		cursor,
+		dayNumberColor,
+		day.Day(),
+		holidayRow,
+		ensureLength(eventRows[0], 11), // 5
+		ensureLength(eventRows[1], 11),
+		ensureLength(eventRows[2], 11),
```

Comment:
This kind of string concatenation I think we need to do in `getCalendarRows()`. Maybe we can return `[]dayRow` instead, a struct that would describe how each row should be rendered. I _think_ that would be easier to manage, that way we could also let `getCalendarRows()` handle start border (`│`), which makes sense and would possible allow for better multi-line event rendering.